### DISCUSSION TITLE
UserProject missing attributes/functions compared to Project

### DISCUSSION THREAD

**Discussion by rickykwanix** [python-gitlab/python-gitlab#2748] [Q&A] [answered]
## Description of the problem, including code/CLI snippet

It is unintuitive for me that I am unable to access `.pipelineschedules()` on a project that came from `myuser.projects.list()`. But the object with the same project id that came from `gl.projects.list(all=True)` has the attribute. I think it'd be more intuitive to have the UserProject interface in the same way as Project. Having the extra API calls to `gl.projects.get(<project_id>)` the projects so that I can access those missing attributes is not ideal.

Maybe this is by design. If so, please help me understand why.

## Expected Behavior

```python
import gitlab
gl = gitlab.Gitlab(...)
user = gl.users.get(123)
user_project = user.projects.list()[0]
print(user_project.id)
print(user_project.pipelineschedules.list(scope="active"))
```

## Actual Behavior

```
4712
Traceback (most recent call last):
  File "/path/to/test.py", line 37, in <module>
    print(user_project.pipelineschedules.list(scope="active"))
  File "/root/.pyenv/versions/3.9.15/lib/python3.9/site-packages/gitlab/base.py", line 134, in __getattr__
    raise AttributeError(message)
AttributeError: 'UserProject' object has no attribute 'pipelineschedules'

<class 'gitlab.v4.objects.users.UserProject'> was created via a list()
call and only a subset of the data may be present. To ensure all data
is present get the object using a get(object.id) call. For more
details, see:

https://python-gitlab.readthedocs.io/en/v4.2.0/faq.html#attribute-error-list
```

## Specifications

  - python-gitlab version: 4.2.0
  - API version you are using (v3/v4): v4
  - Gitlab server version (or gitlab.com): 15.11.3

**Comment by JohnVillalovos** [comment, marked as answer]
> Having the extra API calls to `gl.projects.get(<project_id>)` the projects so that I can access those missing attributes is not ideal.

You might try: https://python-gitlab.readthedocs.io/en/main/api-usage.html#lazy-objects

```
project = gl.projects.get(<project_id>, lazy=True)
project.pipelineschedules.list(scope="active")
```

**Comment by rickykwanix** [comment]
Thanks for converting this into a discussion, I forget that Github has this feature.

And also thanks for letting me know about lazy, I'll make use of that for now.

In terms of Gitlab itself, do they have a distinction/nuance between a project in a user namespace vs non-user namespace?

**Reply by JohnVillalovos** [reply]
If you compare the return values from: https://docs.gitlab.com/ee/api/projects.html#list-all-projects vs https://docs.gitlab.com/ee/api/projects.html#list-user-projects

There are some slight differences.
